module/zfs/arc.c

```diff
@@ -3374,14 +3378,22 @@ arc_convert_to_raw(arc_buf_t *buf, uint64_t dsobj, boolean_t byteorder,
     const uint8_t *mac)
 {
 	arc_buf_hdr_t *hdr = buf->b_hdr;
+	boolean_t need_transfer = B_TRUE;
 
 	ASSERT(ot == DMU_OT_DNODE || ot == DMU_OT_OBJSET);
 	ASSERT(HDR_HAS_L1HDR(hdr));
 	ASSERT3P(hdr->b_l1hdr.b_state, ==, arc_anon);
 
+	if (arc_buf_is_shared(buf)) {
```

Comment:
I don't quite understand this change.  It doesn't seem directly related to the refcount tag.  When do we get passed a shared buf, and why do we need to unshare it?